### PR TITLE
Reorders package managers for python ecosystem metric collection

### PR DESCRIPTION
**Pull request by sachin-sandhu**
### What are you trying to accomplish?

We are not able to see `pipenv` package manager in metric collection, experimenting if reordering package manager as per `update checker` resolves this issue. 

Adds more telemetry data to verify python version that is coming up in metric.

### Checklist

- [x] I have run the complete test suite to ensure all tests and linters pass.
- [x] I have thoroughly tested my code changes to ensure they work as expected, including adding additional tests for new functionality.
- [x] I have written clear and descriptive commit messages.
- [x] I have provided a detailed description of the changes in the pull request, including the problem it addresses, how it fixes the problem, and any relevant details about the implementation.
- [x] I have ensured that the code is well-documented and easy to understand.

### CODE REVIEW

File: python/lib/dependabot/python/file_parser.rb [hunk 1]
```diff
@@ -93,12 +93,12 @@ def package_manager
       def detected_package_manager
         setup_python_environment if Dependabot::Experiments.enabled?(:enable_file_parser_python_local)
 
+        return PipenvPackageManager.new(T.must(detect_pipenv_version)) if detect_pipenv_version
```
Comment (abdulapopoola) [inline python/lib/dependabot/python/file_parser.rb:96]: can we add logs here too to know what is actually detected?

File: python/lib/dependabot/python/file_parser.rb [hunk 2]
```diff
@@ -93,12 +93,12 @@ def package_manager
...
+
         return PoetryPackageManager.new(T.must(detect_poetry_version)) if detect_poetry_version
```
Comment (abdulapopoola) [inline python/lib/dependabot/python/file_parser.rb:98]: [seems the detection based on the package manager names in these files might be the reason?](https://github.com/dependabot/dependabot-core/blob/main/python/lib/dependabot/python/package_manager.rb#L53)

**Review by kbukum1 — APPROVED** [review]
LGTM

File: python/lib/dependabot/python/file_parser.rb (hunk 2 above)
Comment (sachin-sandhu) [reply python/lib/dependabot/python/file_parser.rb:98]: for python , we are evaluating package managers based on the lock file names generated, i.e. poetry.lock , requirements.txt etc,

**Pull request merged**